homeassistant/components/light/knx.py

```diff
@@ -157,17 +153,18 @@ def supported_features(self):
         if self.device.supports_brightness:
             flags |= SUPPORT_BRIGHTNESS
         if self.device.supports_color:
-            flags |= SUPPORT_RGB_COLOR
+            flags |= SUPPORT_COLOR
         return flags
 
     async def async_turn_on(self, **kwargs):
         """Turn the light on."""
         if ATTR_BRIGHTNESS in kwargs:
             if self.device.supports_brightness:
                 await self.device.set_brightness(int(kwargs[ATTR_BRIGHTNESS]))
-        elif ATTR_RGB_COLOR in kwargs:
+        elif ATTR_HS_COLOR in kwargs:
```

Comment:
It seems strange with the `elif` here. We should be able to set brightness and color at the same time.

Reply:
Again, ignoring the brightness in the RGB argument means that there is now no way to set color+brightness in a single service call.

Instead, one must make a separate service call to adjust the brightness, potentially giving a short flash of full brightness until the second call executes.

IMO this is not just maintaining a quirk, it is adding one.

Reply:
@amelchio : do you think something like this would be better:

```python
        if ATTR_BRIGHTNESS in kwargs:
            [change brightness]
        if ATTR_HS_COLOR in kwargs:
            [change color]
        if not ATTR_BRIGHTNESS in kwargs and \
            not ATTR_HS_COLOR in kwargs:
            [switch on]
```

That would change the logic slightly - but i dont know if this is really an actual usecase to get a color **and** brightness change within the same request.

The reason for this awkward logic is that some knx devices flip to 100% brightness when receiving a "set on" telegram.

Reply:
The logic will depend on how the device reacts to the `set_color` call. Does it adjust brightness differently for [100,0,0] vs [200,0,0] or does it just change the color to red (at current brightness)?

Reply:
my guess would be: every manufacturer of KNX devices does it slightly different :)

Reply:
Heh, in that case things could get a bit ugly. I think the plan is to have testers report regressions so maybe just wait and see if any issues pop up?

It definitely is a use case to set color and brightness at the same time, for automations I would even expect it to be quite normal.

Reply:
I will keep an eye on it.